ArduCopter/heli.cpp

```diff
@@ -139,17 +139,16 @@ void Copter::heli_update_rotor_speed_targets()
 
     // get rotor control method
     uint8_t rsc_control_mode = motors->get_rsc_mode();
-
-    float rsc_control_deglitched = rotor_speed_deglitch_filter.apply((float)RC_Channels::rc_channel(CH_8)->get_control_in()) * 0.001f;
-
+    float rsc_control_deglitched = 0.0f;
+    if (rc().find_channel_for_option(RC_Channel::aux_func::MOTOR_INTERLOCK) != nullptr) {
```

Comment:
It would be slightly better to avoid the duplicate calls to rc().find_channel_for_option by storing the return value to a local variable.  It's not hugely important but it saves a small amount of computing power and removes the very unlikely possibility that the 2nd call could return a nullptr leading to an attempted read of uninitialised memory.

Reply:
We should keep the check but we can do it more like this:

```
RC_Channel *rc_ptr = rc().find_channel_for_option(RC_Channel::aux_func::MOTOR_INTERLOCK);
if (rc_ptr != nullptr) {
    rsc_control_deglitched = rotor_speed_deglitch_filter.apply(rc_ptr->get_control_in())*0.001f;
```

Reply:
Ok. I understand. Thanks!